gen2-record-replay/record.py

```diff
@@ -51,53 +53,60 @@ def check_sync(queues, timestamp):
+            # Create recording object for this device
+            recording = Record(str(save_path), device)
+            # Set recording configuration
+            # TODO: add support for specifying resolution, encoding quality
+            recording.set_fps(args.fps)
+            recording.set_save_streams(args.save)
+            recording.start_recording()
 
-        recordings.append(recording)
+            recordings.append(recording)
 
-    queues = [q for recording in recordings for q in recording.queues]
-    while True:
-        for q in queues:
-            new_msg = q['q'].tryGet()
-            if new_msg is not None:
-                q['msgs'].append(new_msg)
-                if check_sync(queues, new_msg.getTimestamp()):
-                    # print('frames synced')
-                    for recording in recordings:
-                        frames = {}
-                        for stream in recording.queues:
-                            frames[stream['name']] = stream['msgs'].pop(0).getCvFrame()
-                            # cv2.imshow(f"{stream['name']} - {device['mx']}", cv2.imdecode(frames[stream['name']], cv2.IMREAD_UNCHANGED))
-                        # print('For mx', device['mx'], 'frames')
-                        # print('frames', frames)
-                        recording.frame_q.put(frames)
-        if cv2.waitKey(1) == ord('q'):
-            break
+        queues = [q for recording in recordings for q in recording.queues]
+        while True:
+            try:
+                for q in queues:
+                    new_msg = q['q'].tryGet()
+                    if new_msg is not None:
+                        q['msgs'].append(new_msg)
+                        if check_sync(queues, new_msg.getTimestamp()):
+                            # print('frames synced')
+                            for recording in recordings:
+                                frames = {}
+                                for stream in recording.queues:
+                                    frames[stream['name']] = stream['msgs'].pop(0).getCvFrame()
+                                    # cv2.imshow(f"{stream['name']} - {device['mx']}", cv2.imdecode(frames[stream['name']], cv2.IMREAD_UNCHANGED))
+                                # print('For mx', device['mx'], 'frames')
+                                # print('frames', frames)
+                                recording.frame_q.put(frames)
+                if cv2.waitKey(1) == ord('q'):
+                    break
+            except KeyboardInterrupt:
+                break
 
-    for recording in recordings:
-        recording.frame_q.put(None)
-        time.sleep(0.01) # Wait 10ms for process to close all video files
-        recording.process.join() # Terminate the process
+        for recording in recordings:
+            recording.frame_q.put(None)
+            time.sleep(0.01) # Wait 10ms for process to close all video files
+            recording.process.join() # Terminate the process
```

Comment:
`join()` actually doesn't terminate the process, just hangs until it finishes - so the sleep before seems unnecessary